src/KeyringController.test.ts

```diff
@@ -527,48 +556,54 @@ describe('KeyringController', function () {
       expect(keyringController.keyrings).toHaveLength(2);
 
       // remove one account from the keyring we just added
-      await keyringController.removeAccount(walletTwoAddresses[0]);
+      await keyringController.removeAccount(walletTwoAddresses[0] as Hex);
 
       // Check that the newly added keyring was not removed after
       // removing the account since it still has an account left
       expect(keyringController.keyrings).toHaveLength(2);
     });
   });
 
-  describe('unlockKeyrings', function () {
-    it('returns the list of keyrings', async function () {
+  describe('unlockKeyrings', () => {
+    it('returns the list of keyrings', async () => {
       await keyringController.setLocked();
       const keyrings = await keyringController.unlockKeyrings(PASSWORD);
       expect(keyrings).toHaveLength(1);
       await Promise.all(
         keyrings.map(async (keyring) => {
-          const wallet = await keyring.serialize();
-          expect(wallet.numberOfAccounts).toBe(1);
+          const { numberOfAccounts } = (await keyring.serialize()) as any;
+          expect(numberOfAccounts).toBe(1);
         }),
       );
     });
 
-    it('add serialized keyring to _unsupportedKeyrings array if keyring type is not known', async function () {
-      const _unsupportedKeyrings = [{ type: 'Ledger Keyring', data: 'DUMMY' }];
-      mockEncryptor.encrypt(PASSWORD, _unsupportedKeyrings);
+    it('add serialized keyring to unsupportedKeyrings array if keyring type is not known', async () => {
+      const unsupportedKeyrings = [{ type: 'Ledger Keyring', data: 'DUMMY' }];
+      mockEncryptor.encrypt(PASSWORD, unsupportedKeyrings);
       await keyringController.setLocked();
       const keyrings = await keyringController.unlockKeyrings(PASSWORD);
       expect(keyrings).toHaveLength(0);
-      expect(keyringController._unsupportedKeyrings).toStrictEqual(
-        _unsupportedKeyrings,
+      expect(keyringController.unsupportedKeyrings).toStrictEqual(
+        unsupportedKeyrings,
       );
     });
   });
 
-  describe('verifyPassword', function () {
+  describe('verifyPassword', () => {
     beforeEach(() => {
       keyringController = new KeyringController({
+        keyringBuilders: [
+          keyringBuilderFactory(
+            KeyringMockWithInit as unknown as KeyringClass<any>,
```

Comment:
Why typecast this? It seems like a mock keyring that we create should conform to the keyring class interface.

Reply:
It should. Done in [977facf](https://github.com/MetaMask/KeyringController/pull/202/commits/977facf11c37f5418c5303954037d7ecf125b734).

I also removed the rest of typecasting in this module.